<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>edit_bookmark</class>
 <widget class="QDialog" name="edit_bookmark">
  <property name="windowModality">
   <enum>Qt::NonModal</enum>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>944</width>
    <height>818</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Edit preset</string>
  </property>
  <property name="modal">
   <bool>false</bool>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_4">
   <item>
    <widget class="QGroupBox" name="group_general">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="title">
      <string>General parameters</string>
     </property>
     <layout class="QFormLayout" name="formLayout_3">
      <item row="0" column="0">
       <widget class="QLabel" name="label_4">
        <property name="toolTip">
         <string>Set the name of the bookmark. Must be unique.</string>
        </property>
        <property name="text">
         <string>Preset name</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="bookmark_name">
        <property name="toolTip">
         <string>Set the name of the bookmark. Must be unique.</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_6">
        <property name="toolTip">
         <string>Description of the bookmark or per query. One description per line.</string>
        </property>
        <property name="text">
         <string>Description(s)</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QPlainTextEdit" name="description">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>100</height>
         </size>
        </property>
        <property name="toolTip">
         <string>Description of the bookmark or per query. One description per line.</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>Preset type</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <property name="spacing">
         <number>10</number>
        </property>
        <property name="leftMargin">
         <number>50</number>
        </property>
        <property name="rightMargin">
         <number>400</number>
        </property>
        <item>
         <widget class="QRadioButton" name="radio_basic">
          <property name="text">
           <string>Basic</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="radio_advanced">
          <property name="text">
           <string>Advanced</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QListWidget" name="list_queries">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="maximumSize">
          <size>
           <width>200</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="palette">
          <palette>
           <active>
            <colorrole role="Base">
             <brush brushstyle="SolidPattern">
              <color alpha="255">
               <red>240</red>
               <green>240</green>
               <blue>240</blue>
              </color>
             </brush>
            </colorrole>
           </active>
           <inactive>
            <colorrole role="Base">
             <brush brushstyle="SolidPattern">
              <color alpha="255">
               <red>240</red>
               <green>240</green>
               <blue>240</blue>
              </color>
             </brush>
            </colorrole>
           </inactive>
           <disabled>
            <colorrole role="Base">
             <brush brushstyle="SolidPattern">
              <color alpha="255">
               <red>240</red>
               <green>240</green>
               <blue>240</blue>
              </color>
             </brush>
            </colorrole>
           </disabled>
          </palette>
         </property>
         <property name="editTriggers">
          <set>QAbstractItemView::NoEditTriggers</set>
         </property>
         <property name="currentRow">
          <number>-1</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="button_add">
         <property name="text">
          <string>Add a new query</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_5">
       <item>
        <widget class="QScrollArea" name="scrollArea">
         <property name="widgetResizable">
          <bool>true</bool>
         </property>
         <widget class="QWidget" name="scrollAreaWidgetContents">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>0</y>
            <width>695</width>
            <height>598</height>
           </rect>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <item>
            <widget class="QGroupBox" name="group_query">
             <property name="title">
              <string>Query parameters</string>
             </property>
             <layout class="QFormLayout" name="formLayout">
              <item row="0" column="0">
               <widget class="QLabel" name="layer_name_label">
                <property name="toolTip">
                 <string>Set a layer name for the query.</string>
                </property>
                <property name="frameShape">
                 <enum>QFrame::NoFrame</enum>
                </property>
                <property name="text">
                 <string>Layer name</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QLineEdit" name="layer_name">
                <property name="toolTip">
                 <string>Set a layer name for the query.</string>
                </property>
               </widget>
              </item>
              <item row="4" column="0">
               <widget class="QLabel" name="label_7">
                <property name="toolTip">
                 <string>Surcharge {{GeocodeArea}}. Name of the place</string>
                </property>
                <property name="text">
                 <string>Area</string>
                </property>
               </widget>
              </item>
              <item row="4" column="1">
               <widget class="QLineEdit" name="area">
                <property name="toolTip">
                 <string>Surcharge {{GeocodeArea}}. Name of the place</string>
                </property>
               </widget>
              </item>
              <item row="5" column="0">
               <widget class="QLabel" name="label_8">
                <property name="toolTip">
                 <string>Surcharge {{bbox}}. Extent (EPSG:4326): xmin ymin xmax ymax</string>
                </property>
                <property name="text">
                 <string>Bbox</string>
                </property>
               </widget>
              </item>
              <item row="5" column="1">
               <layout class="QHBoxLayout" name="bbox_layout"/>
              </item>
              <item row="2" column="1">
               <widget class="QStackedWidget" name="stacked_parameters_bookmark">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="currentIndex">
                 <number>1</number>
                </property>
                <widget class="QWidget" name="advanced_parameters">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <layout class="QVBoxLayout" name="verticalLayout_6">
                  <property name="leftMargin">
                   <number>0</number>
                  </property>
                  <property name="topMargin">
                   <number>0</number>
                  </property>
                  <property name="rightMargin">
                   <number>0</number>
                  </property>
                  <property name="bottomMargin">
                   <number>0</number>
                  </property>
                  <item>
                   <widget class="QPlainTextEdit" name="query">
                    <property name="sizePolicy">
                     <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                      <horstretch>0</horstretch>
                      <verstretch>0</verstretch>
                     </sizepolicy>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </widget>
                <widget class="QWidget" name="basic_parameters">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <layout class="QVBoxLayout" name="verticalLayout_7">
                  <property name="leftMargin">
                   <number>0</number>
                  </property>
                  <property name="topMargin">
                   <number>0</number>
                  </property>
                  <property name="rightMargin">
                   <number>0</number>
                  </property>
                  <property name="bottomMargin">
                   <number>0</number>
                  </property>
                  <item>
                   <layout class="QVBoxLayout" name="verticalLayout_18">
                    <item>
                     <layout class="QHBoxLayout" name="horizontalLayout_12">
                      <item>
                       <widget class="QLabel" name="label_41">
                        <property name="sizePolicy">
                         <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                          <horstretch>0</horstretch>
                          <verstretch>0</verstretch>
                         </sizepolicy>
                        </property>
                        <property name="text">
                         <string>Preset</string>
                        </property>
                       </widget>
                      </item>
                      <item>
                       <widget class="QComboBox" name="combo_preset_eb">
                        <property name="sizePolicy">
                         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                          <horstretch>0</horstretch>
                          <verstretch>0</verstretch>
                         </sizepolicy>
                        </property>
                        <property name="toolTip">
                         <string>A preset to fetch</string>
                        </property>
                        <property name="editable">
                         <bool>true</bool>
                        </property>
                       </widget>
                      </item>
                     </layout>
                    </item>
                    <item>
                     <widget class="QTableWidget" name="table_keys_values_eb">
                      <property name="sizePolicy">
                       <sizepolicy hsizetype="Expanding" vsizetype="Ignored">
                        <horstretch>0</horstretch>
                        <verstretch>0</verstretch>
                       </sizepolicy>
                      </property>
                      <row>
                       <property name="text">
                        <string>1</string>
                       </property>
                      </row>
                      <column>
                       <property name="text">
                        <string/>
                       </property>
                      </column>
                      <column>
                       <property name="text">
                        <string>Key</string>
                       </property>
                      </column>
                      <column>
                       <property name="text">
                        <string>Value</string>
                       </property>
                      </column>
                      <column>
                       <property name="text">
                        <string>Add</string>
                       </property>
                      </column>
                      <column>
                       <property name="text">
                        <string>Delete</string>
                       </property>
                      </column>
                     </widget>
                    </item>
                   </layout>
                  </item>
                 </layout>
                </widget>
               </widget>
              </item>
              <item row="2" column="0">
               <widget class="QLabel" name="query_parameter">
                <property name="text">
                 <string>Query</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QGroupBox" name="group_output">
             <property name="title">
              <string>Output parameters</string>
             </property>
             <layout class="QFormLayout" name="formLayout_2">
              <item row="0" column="0">
               <widget class="QLabel" name="label_2">
                <property name="text">
                 <string>Output geometry</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QLabel" name="label_3">
                <property name="text">
                 <string>White list of columns</string>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QCheckBox" name="checkbox_points">
                <property name="text">
                 <string>Points</string>
                </property>
               </widget>
              </item>
              <item row="1" column="1">
               <widget class="QLineEdit" name="white_points"/>
              </item>
              <item row="2" column="0">
               <widget class="QCheckBox" name="checkbox_lines">
                <property name="text">
                 <string>Lines</string>
                </property>
               </widget>
              </item>
              <item row="2" column="1">
               <widget class="QLineEdit" name="white_lines"/>
              </item>
              <item row="3" column="0">
               <widget class="QCheckBox" name="checkbox_multilinestrings">
                <property name="text">
                 <string>Multilinestrings</string>
                </property>
               </widget>
              </item>
              <item row="3" column="1">
               <widget class="QLineEdit" name="white_multilinestrings"/>
              </item>
              <item row="4" column="0">
               <widget class="QCheckBox" name="checkbox_multipolygons">
                <property name="text">
                 <string>Multipolygons</string>
                </property>
               </widget>
              </item>
              <item row="4" column="1">
               <widget class="QLineEdit" name="white_multipolygons"/>
              </item>
              <item row="5" column="0">
               <widget class="QLabel" name="outputFormatLabel">
                <property name="text">
                 <string>Output format</string>
                </property>
               </widget>
              </item>
              <item row="5" column="1">
               <widget class="QComboBox" name="combo_output_format"/>
              </item>
              <item row="6" column="0">
               <widget class="QLabel" name="label">
                <property name="text">
                 <string>Output directory</string>
                </property>
               </widget>
              </item>
              <item row="6" column="1">
               <widget class="QgsFileWidget" name="output_directory" native="true"/>
              </item>
              <item row="7" column="1">
               <widget class="QLabel" name="label_9">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="text">
                 <string>To associate a qml style file for each layer, you must add this file in the folder QuickOSM/'name of the preset' with the name in a peculiar format : 'name of the preset'_'name of the query'_'layer'.</string>
                </property>
                <property name="wordWrap">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QPushButton" name="button_validate">
           <property name="text">
            <string>Validate</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="button_cancel">
           <property name="text">
            <string>Cancel</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QgsFileWidget</class>
   <extends>QWidget</extends>
   <header>qgsfilewidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
